File: 151001/Shcherba/lab5/publicator/src/main/java/by/bsuir/dto/TweetRequestTo.java
package by.bsuir.dto;

import by.bsuir.utils.StringDeserializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import jakarta.validation.constraints.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TweetRequestTo {
    private Long id;
    @NotBlank
    @Size(min = 2, max = 64)
    private String title;
    @JsonDeserialize(using = StringDeserializer.class)
    private String content;
    private Long userId;
    private Long markerId;
}

File: 151004/Golovchuk/Lab5/publisher/DC_Lab1/DTO/TweetRequestTo.cs
using DC_Lab1.DTO.Interface;
using DC_Lab1.Models;

namespace DC_Lab1.DTO
{
    public class TweetRequestTo(int Id, int? creatorId, string? Title, string? Content) : IRequestTo
    {
        public int Id { get; set; } = Id;

        public int? creatorId { get; set; } = creatorId;

        public string? Title { get; set; } = Title;

        public string? Content { get; set; } = Content;


    }
}
